.github/workflows/build-fluentbit-image.yaml

```
@@ -0,0 +1,261 @@
+        id: determine-tags
+        run: |
+          VERSION=$(cat cmd/fluent-watcher/fluentbit/VERSION)  # Read the version from the VERSION file
+          VERSION_WITHOUT_V=${VERSION#v}
+          MAJOR_MINOR=$(echo $VERSION_WITHOUT_V | cut -d. -f1-2)
+
+          if skopeo inspect docker://${{ env.GHCR_REPO }}:${VERSION}; then
+            echo "${VERSION} tag already exists, assuming we're building a patch release!"
+            LATEST_PATCH_VERSION=$(skopeo list-tags docker://${{ env.GHCR_REPO }} | grep -E "${VERSION}-[0-9]+" | sort | uniq | tail -1 | tr -d \" | cut -d'-' -f2 | tr -d ',')
+            NEW_PATCH_VERSION=$((LATEST_PATCH_VERSION + 1))
+            IMAGE_BASE_TAG="${VERSION}-${NEW_PATCH_VERSION}"
+            echo "Building patch release ${IMAGE_BASE_TAG}!"
+          else
+            echo "${VERSION} tag does not exist, assuming we're building a new release!"
+            IMAGE_BASE_TAG="${VERSION}"
+          fi
+  
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+          echo "image-base-tag=$IMAGE_BASE_TAG" >> $GITHUB_OUTPUT
+          echo "image-major-minor=$MAJOR_MINOR" >> $GITHUB_OUTPUT
+
+  build-tags:
+    name: Get image tags
+    runs-on: ubuntu-latest
+    needs: [build-info]
+
+    outputs:
+      tags: ${{ steps.image-metadata.outputs.tags }}
+      labels: ${{ steps.image-metadata.outputs.labels }}
+      debug-tags: ${{ steps.debug-image-metadata.outputs.tags }}
+      debug-labels: ${{ steps.debug-image-metadata.outputs.labels }}
+
+    steps:
+      - name: Configure image tags 
+        id: image-metadata
+        uses: docker/metadata-action@902fa8ec7d6ecbf8d84d538b9b233a880e428804 # v5.7.0
+        with:
+          images: |
+            ${{ env.GHCR_REPO }}
+          flavor:
+            latest=false
+          tags: |
+            type=raw,value=latest
+            type=raw,value=${{ needs.build-info.outputs.image-base-tag }}
+            type=raw,value=v${{ needs.build-info.outputs.image-base-tag }}
+            type=raw,value=${{ needs.build-info.outputs.image-major-minor }}
+            type=raw,value=v${{ needs.build-info.outputs.image-major-minor }}   
+
+      - name: Configure debug image tags 
+        id: debug-image-metadata
+        uses: docker/metadata-action@902fa8ec7d6ecbf8d84d538b9b233a880e428804 # v5.7.0
+        with:
+          images: |
+            ${{ env.GHCR_REPO }}
+            ${{ env.DOCKERHUB_REPO }}
+          flavor:
+            suffix=-debug
+          tags: |
+            type=raw,value=${{ needs.build-info.outputs.image-base-tag }}
+            type=raw,value=v${{ needs.build-info.outputs.image-base-tag }}
+            type=raw,value=${{ needs.build-info.outputs.image-major-minor }}
+            type=raw,value=v${{ needs.build-info.outputs.image-major-minor }}                 
+
+  build:
+    name: Build/push image (${{ matrix.platform }})
+    runs-on: ${{ matrix.runs-on }}
+    needs: [build-info,build-tags]
+    permissions:
+      actions: read
```

```suggestion
      contents: read
```

Not sure why you need `actions: read`? Did you intended to use `contents: read`?